electron/src/updater/windowsUpdater.ts

```diff
@@ -0,0 +1,125 @@
+"use strict";
+
+import * as Promise from "bluebird";
+import { spawn } from "child_process";
+import { app } from "electron";
+import { EventEmitter } from "events";
+import * as fs from "fs";
+import * as I from "../typings/interfaces";
+import * as path from "path";
+import * as Utils from "../util/electronUtils";
+
+export class WindowsUpdater extends EventEmitter implements I.UpdaterBase {
+    private url: string = null;
+    private currentRequest: Promise<void> = null;
+    private updatePackagePath: string = null;
+
+    constructor(private cacheInfo: I.UpdateCacheInfo) {
+        super();
+    }
+
+    public setFeedURL(url: string): void {
+        this.url = url;
+    }
+
+    public checkForUpdates(): void {
+        if (!this.url) {
```

Comment:
This file is lacking in comments. Not a blocker for merging the PR. But please add it after merging.